python/cudf/cudf/core/index.py

```diff
@@ -591,48 +591,43 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         )
 
     @_cudf_nvtx_annotate
-    def get_loc(self, key, method=None, tolerance=None):
-        # We should not actually remove this code until we have implemented the
-        # get_indexers method as an alternative, see
-        # https://github.com/rapidsai/cudf/issues/12312
-        if method is not None:
-            warnings.warn(
-                f"Passing method to {self.__class__.__name__}.get_loc is "
-                "deprecated and will raise in a future version.",
-                FutureWarning,
+    def get_indexer(self, target, limit=None, method=None, tolerance=None):
+        if method is None:
+            if self.step > 0:
+                start, stop, step = self.start, self.stop, self.step
+            else:
+                # Reversed
+                reverse = self._range[::-1]
+                start, stop, step = reverse.start, reverse.stop, reverse.step
+
+            target_array = cupy.asarray(target)
```

Comment:
In pandas, `RangeIndex.get_loc(["a", "b"])` just returns `[-1, -1]` (because those labels aren't in the index). Here we will get a runtime failure. We should check the dtype somehow (argh!) and return a full array of -1 if things don't match.

Reply:
Fixed it 👍